lib/created_message_queue_handler.ts

```diff
@@ -112,34 +111,73 @@ type ContextWithBindings = t.TypeOf<typeof ContextWithBindings> & IContext;
  * @param maybeProfile      the recipient's profile (or none)
  * @param defaultAddresses  default addresses (one per channel)
  */
-function getEmailNotification(
+function tryGetDestinationEmail(
   maybeProfile: Option<RetrievedProfile>,
   defaultAddresses: Option<NewMessageDefaultAddresses>
-): Option<NotificationChannelEmail> {
+): Option<Tuple2<EmailAddress, NotificationAddressSourceEnum>> {
   const maybeProfileEmail = maybeProfile
     .chain(profile => fromNullable(profile.email))
     .map(email => Tuple2(email, NotificationAddressSourceEnum.PROFILE_ADDRESS));
   const maybeDefaultEmail = defaultAddresses
     .chain(addresses => fromNullable(addresses.email))
     .map(email => Tuple2(email, NotificationAddressSourceEnum.DEFAULT_ADDRESS));
-  const maybeEmail = maybeProfileEmail.alt(maybeDefaultEmail);
-  return maybeEmail.map(({ e1: toAddress, e2: addressSource }) => {
+  return maybeProfileEmail.alt(maybeDefaultEmail);
+}
+
+/**
+ * Attempt to resolve an email notification.
+ *
+ * @param maybeProfile the user's profile we try to get an email address from
+ * @param defaultAddresses the email address provided as input to the message APIs
+ * @returns none when both maybeProfile.email and defaultAddresses are empty
+ */
+function tryGetEmailNotification(
+  maybeProfile: Option<RetrievedProfile>,
+  defaultAddresses: Option<NewMessageDefaultAddresses>
+): Option<NotificationChannelEmail> {
+  const maybeDestinationEmail = tryGetDestinationEmail(
+    maybeProfile,
+    defaultAddresses
+  );
+  // Now that we have a valid destination email address, we create an EmailNotification
+  return maybeDestinationEmail.map(({ e1: toAddress, e2: addressSource }) => {
     return {
       addressSource,
-      status: NotificationChannelStatusEnum.QUEUED,
       toAddress
     };
   });
 }
 
+/**
+ * Save a notification object to the database.
+ *
+ * @param notificationModel notification model used to interact with database
+ * @param notifications     array of notification objects
+ */
+async function saveNotification(
+  notificationModel: NotificationModel,
+  notification: NewNotification
+): Promise<Either<RuntimeError, RetrievedNotification>> {
```

Comment:
since this always errors with `TransientError` you can be more specific here and return `Promise<Either< TransientError, RetrievedNotification>>` (if possible, it's always good to hint the compiler about more specific types)